src/plugins/data_source/server/routes/test_connection.ts

```diff
@@ -9,11 +9,13 @@ import { AuthType, DataSourceAttributes, SigV4ServiceName } from '../../common/d
 import { DataSourceConnectionValidator } from './data_source_connection_validator';
 import { DataSourceServiceSetup } from '../data_source_service';
 import { CryptographyServiceSetup } from '../cryptography_service';
+import { IAuthenticationMethodRegistery } from '../auth_registry';
 
 export const registerTestConnectionRoute = (
   router: IRouter,
   dataSourceServiceSetup: DataSourceServiceSetup,
-  cryptography: CryptographyServiceSetup
+  cryptography: CryptographyServiceSetup,
+  authRegistryPromise: Promise<IAuthenticationMethodRegistery>
```

Comment:
Since we update "dataSourceAttr" to support "AuthTypeContent". Do we plan to also update the "dataSourceAttr" here in validate object?

Reply:
Yes. This PR only covers adding interfaces. I will cover usage of auth registry and refactoring of existing code in following PR.